libraries/ms-common/src/stm32f0xx/interrupt_mcu.c

```diff
@@ -0,0 +1,92 @@
+#include "interrupt_mcu.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "interrupt.h"
+#include "status.h"
+#include "stm32f0xx_exti.h"
+#include "stm32f0xx_misc.h"
+#include "stm32f0xx_rcc.h"
+#include "stm32f0xx_syscfg.h"
+
+// Channels are NVIC channels which handle prioritization of interupts.
+#define NUM_INTERRUPT_MCU_CHANNELS 32
+// Lines are external interrupt channels often grouped to a single
+#define NUM_INTERRUPT_MCU_LINES 32
+
+static InterruptPriority s_interrupt_mcu_priorities[NUM_INTERRUPT_MCU_CHANNELS];
+static bool s_interrupt_mcu_initialized = false;
+
+void interrupt_mcu_init(void) {
+  if (!s_interrupt_mcu_initialized) {
```

Comment:
Done. Required an `interrupt_init(void)` function to be added to the `interrupt.h` file which needs to be called to start interrupts on any given platform. This was because I was planning to call the `<platform>_interrupt_init(void)` in the specific interrupt module init functions ie. `gpio_it_init()`, `timer_it_init`, etc. and that would have unregistered interrupts for the other modules.